libraries/DataFlash/DataFlash_Backend.cpp

```diff
@@ -260,6 +260,31 @@ bool DataFlash_Backend::Log_Write(const uint8_t msg_type, va_list arg_list, bool
     return WritePrioritisedBlock(buffer, msg_len, is_critical);
 }
 
+bool DataFlash_Backend::StartNewLogOK() const
+{
+    if (logging_started()) {
+        return false;
+    }
+    if (_front._log_bitmask == 0) {
+        return false;
+    }
+    if (_front.in_log_download()) {
+        return false;
+    }
+    if (!DataFlash_Backend::WritesOK()) {
```

Comment:
I was thinking of splitting the code out of WritesOK into a separate method which was called by WritesOK and by StartNewLogOK.  I couldn't come up with a name I liked.  So I just called the other one.  I'll come up with a name.  It's be non-virtual and private, so it doesn't matter *too* much....